src/index.ts

```diff
@@ -44,8 +44,8 @@ export function createNetworkInterface(
 
 function extractFiles(variables: object): { variables: object; files: Files } {
     const files: Files = [];
-    const walkTree = (tree: object, path: string[] = []): object => {
-        const mapped = { ...tree };
+    const walkTree = (tree: any, path: string[] = []): object => {
+        const mapped = isArray(tree) ? [...tree] : { ...tree };
```

Comment:
Please use [`Array.isArray`][1] instead of your self-rolled `isArray`-solution.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray